src/Peachpie.Library/Hash.cs

```diff
@@ -3024,13 +3032,422 @@ public static string crypt(string str, string salt/* mandatory since 5.6 */)
             else
             {
                 // DES
-                throw new NotImplementedException();
+                return DES.Crypt(str, salt);
             }
 
             // failure
             return salt.StartsWith("*0") ? "*1" : "*0";
         }
 
+        private static string ConvertByteArrayToString(byte[] buffer)
+        {
+            StringBuilder builder = new StringBuilder();
+            foreach (byte value in buffer)
+            {
+                builder.Append((char)value);
+            }
+            return builder.ToString().Trim('\0');
+        }
+
+        #region md5 hash in crypt function
+        // This code in region was copied and modified from The PHP Interpreter (https://github.com/php/php-src/blob/master/ext/standard/md5.c) 
+        private const int MD5MaxLength = 120;
+        private const string MD5Name = "md5";
+        private const string MD5Magic = "$1$";
+        private const string itoa64 =     /* 0 ... 63 => ascii - 64 */
+        "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+
+        /// <summary>
+        /// Internal funcion of php to map hash symbols to itoa64 set.
+        /// </summary>
+        /// <param name="hash"></param>
+        /// <param name="s">Position, where mapping starts.</param>
+        /// <param name="v"></param>
+        /// <param name="n">Length of mapping sequence.</param>
+        private static void MD5MapASCII(byte[] hash, int s, int v, int n)
+        {
+            while (--n >= 0)
+            {
+                hash[s++] = (byte)itoa64[v & 0x3f];
+                v >>= 6;
+            }
+        }
+
+        /// <summary>
+        /// Internal funcion of php to map hash symbols to itoa64 set.
+        /// </summary>
+        /// <param name="hash"></param>
+        /// <param name="s">Position, where mapping starts.</param>
+        /// <param name="n">Length of mapping sequence.</param>
+        private static void MD5MapASCII(byte[] hash, int n)
+        {
+            int s = 0;
+            while (--n >= 0)
+            {
+                hash[s] = (byte)itoa64[hash[s] & 0x3f];
+                s++;
+            }
+        }
+
+        /// <summary>
+        /// Part of cypt function in php. 
+        /// </summary>
+        private static string CryptMD5(Context ctx, string password, string salt)
```

Comment:
`Context ctx` is not needed.

Reply:
Resolved https://github.com/peachpiecompiler/peachpie/pull/593/commits/6f1865f58f6166fda85c7352d98cab2d049e5ddf